src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -701,7 +701,8 @@ private MemberSemanticModel GetMemberModel(int position)
                         var constructorDecl = (ConstructorDeclarationSyntax)memberDecl;
                         outsideMemberDecl =
                             !LookupPosition.IsInConstructorParameterScope(position, constructorDecl) &&
-                            !LookupPosition.IsInParameterList(position, constructorDecl);
+                            !LookupPosition.IsInParameterList(position, constructorDecl) &&
+                            !LookupPosition.IsInExpressionBody(position, constructorDecl.GetExpressionBodySyntax(), constructorDecl.SemicolonToken);
```

Comment:
This doesn't feel like the right place for this condition. I think implementation of LookupPosition.IsInConstructorParameterScope should be adjusted instead to properly handle an expression body.